ofrak_core/ofrak/core/uefi.py

```diff
@@ -0,0 +1,53 @@
+import os
+import asyncio
+import logging
+import tempfile
+from dataclasses import dataclass
+from subprocess import CalledProcessError
+
+from ofrak.component.unpacker import Unpacker
+from ofrak.resource import Resource
+from ofrak.core.filesystem import File, Folder, FilesystemRoot, SpecialFileType
+
+
+from ofrak.model.component_model import ComponentExternalTool
+from ofrak.core.pe.model import Pe
+
+LOGGER = logging.getLogger(__name__)
+
+UEFIEXTRACT = ComponentExternalTool("uefiextract", "https://github.com/LongSoft/UEFITool", "--help")
+
+
+@dataclass
+class Uefi(FilesystemRoot, Pe):
+    """
+    Filesystem extracted from a UEFI binary.
+    """
+
+
+class UefiUnpacker(Unpacker[None]):
+    """Unpack a UEFI binary. This current method cannot repack after modification because UEFITool cannot support it."""
+
+    targets = (Uefi,)
+    children = (File, Folder, SpecialFileType)
+    external_dependencies = (UEFIEXTRACT,)
+
+    async def unpack(self, resource: Resource, config=None):
+        ROM_FILE = "uefi.rom"
+
+        with tempfile.TemporaryDirectory() as temp_flush_dir:
+            os.chdir(temp_flush_dir)
```

Comment:
Add a comment noting why the `chdir` is necessary – don't want someone removing it in the future without knowing why it was put in. 

In general, code comments that explain what something does are not useful. But code comments that explain why something was done are extremely useful.